src/Uuid/Uuid.php

```diff
@@ -0,0 +1,334 @@
+			$seed instanceof Page
+				=> new PageUuid(model: $seed, context: $context),
+			$seed instanceof File
+				=> new FileUuid(model: $seed, context: $context),
+			$seed instanceof Site
+				=> new SiteUuid(model: $seed, context: $context),
+			$seed instanceof User
+				=> new UserUuid(model: $seed, context: $context),
+			// TODO: activate for uuid-block-structure-support
+			// $seed instanceof Block
+			// 	=> new BlockUuid(model: $seed, context: $context),
+			// $seed instanceof StructureObject
+			// 	=> new StructureUuid(model: $seed, context: $context),
+			default
+			=> throw new InvalidArgumentException('UUID not supported for: ' . get_class($seed))
+		};
+	}
+
+	/**
+	 * Generate a new ID string
+	 */
+	final public static function generate(int $length = 15): string
+	{
+		if (static::$generator !== null) {
+			return (static::$generator)($length);
+		}
+
+		return Str::random($length);
+	}
+
+	/**
+	 * Returns the UUID's id string
+	 */
+	public function id(): string
+	{
+		return $this->uri->host();
+	}
+
+	/**
+	 * Generator function that creates an index of
+	 * all identifiable model object globally
+	 */
+	public static function index(): Generator
+	{
+		yield from [];
+	}
+
+	/**
+	 * Merges local and global index generators
+	 * into one iterator
+	 * @internal
+	 *
+	 * @return \Generator|\Kirby\Uuid\Identifiable[]
+	 */
+	final public function indexes(): Generator
+	{
+		yield from $this->context();
+		yield from static::index();
+	}
+
+	/**
+	 * Checks if a string resembles an UUID uri
+	 */
+	final public static function is(
+		string $string,
+		string|null $type = null
+	): bool {
+		if ($type !== null) {
+			return Str::startsWith($string, $type . '://');
+		}
+
+		// try to match any of the supported schemes
+		$pattern = sprintf('/^(%s):\/\//', implode('|', Uri::$schemes));
+		return preg_match($pattern, $string) === 1;
+	}
+
+	/**
+	 * Checks if Uuid has already been cached
+	 */
+	public function isCached(): bool
+	{
+		return Uuids::cache()->exists($this->key());
+	}
+
+	/**
+	 * Returns key for cache entry
+	 */
+	public function key(): string
+	{
+		$id = $this->id();
+
+		// for better performance when using a file-based cache,
+		// turn first two characters of the id into a directory
+		$id = Str::substr($id, 0, 2) . '/' . Str::substr($id, 2);
+
+		return static::TYPE . '/' . $id;
+	}
+
+	/**
+	 * Feeds Uuid into the cache
+	 *
+	 * @return bool
+	 */
+	public function populate(): bool
+	{
+		return Uuids::cache()->set($this->key(), $this->value());
+	}
+
+	/**
+	 * Returns the full UUID string incl. scheme
+	 */
+	public function render(): string
+	{
+		// make sure id is generated if
+		// it doesn't exist yet
+		$this->id();
+
+		return $this->uri->toString();
+	}
+
+	/**
+	 * Tries to find the identifiable model in cache
+	 * or index and returns the object
+	 *
+	 * @param bool $lazy If `true`, only lookup from cache
+	 */
+	public function resolve(bool $lazy = false): Identifiable|null
```

Comment:
A lot of the upcoming blueprint refactoring by @bastianallgeier is structured along resolve/render which is why I thought it could be nice to stick to that terminology here as well.

Reply:
OK, I see. Leaving this discussion open for a second opinion from the team. Anyone can feel free to resolve it.

Reply:
Both names work for me tbh.

Reply:
What I like about `model()` is that it matches the prop name, like we do it in other places.

Reply:
Add model() as alias?
I don't think render would make much sense on its own.

Reply:
An alias would make it even more confusing TBH. What if we rename render to toString?

Reply:
Not sure, would be a good chance to use the same terminology here that we will use so much in the future. Probably @bastianallgeier needs to make a decision even if he likes both 😄

Reply:
I think the new stuff is still very much under construction. There are a few things that already seem very useful, but I'm not 100% sure about how "final" render and resolve really are. Maybe that's already a good indicator to actually stick with model instead.

Reply:
Giving in to the majority - made the changes.